cvmfs/history.py

```diff
@@ -54,9 +70,24 @@ def __iter__(self):
         return self.list_tags().__iter__()
 
     def list_tags(self):
-        results = self.run_sql(RevisionTag.sql_query())
+        results = self.run_sql(RevisionTag.sql_query_all())
         return [ RevisionTag(sql_res) for sql_res in results ]
 
+    def get_tag_by_name(self, tag_name):
+        result = self.run_sql(RevisionTag.sql_query_name(tag_name))
+        if result:
+            return RevisionTag(result[0])
+
+    def get_tag_by_revision(self, revision):
+        result = self.run_sql(RevisionTag.sql_query_revision(revision))
+        if result:
+            return RevisionTag(result[0])
+
+    def get_tag_by_date(self, timestamp):
+        result = self.run_sql(RevisionTag.sql_query_date(timestamp))
+        if result:
+            return RevisionTag(result[0])
+
```

Comment:
Let's DRY those up a bit like this:
```python
def _get_tag_by_query(self, query):
  result = self.run_sql(query)
  if result:
    return RevisionTag(result[0])
```

The others should then just call into `_get_tag_by_query()` with the appropriate query string.